lib/lint.js

```diff
@@ -18,16 +18,18 @@ export default function lint(editor) {
 
 	const cwd = path.dirname(filename);
 
-	const pendingContext = pProps({
-		cwd: getPackagePath(cwd).then(pkgPath => pkgPath ? path.dirname(pkgPath) : cwd),
-		depends: dependsOnXO(cwd),
-		xo: getXO(cwd)
-	});
+	const pendingContext = getPackagePath(cwd)
+		.then(pkgPath => pProps({
+			cwd: pkgPath ? path.dirname(pkgPath) : cwd,
+			depends: dependsOnXO(cwd),
+			xo: getXO(cwd),
+			xoSettings: pkgPath ? getPackageData(pkgPath).then(pkg => pkg.xo || {}) : {}
```

Comment:
This should not be necessary. `xo.lintText()` already picks up options from package.json.

Reply:
Dunno why it didn't work for me before. After removing node_modules, `npm i` and a brief debugging session it "just works".

I've removed that commit.